hydra/_internal/config_loader_impl.py

```diff
@@ -225,6 +225,13 @@ def _load_configuration_impl(
 
         self._process_config_searchpath(config_name, parsed_overrides, caching_repo)
 
+        for source in caching_repo.get_sources():
+            if not source.available():
+                warnings.warn(
+                    category=UserWarning,
+                    message=f"provider={source.provider}, path={source.path} is not available.",
+                )
+
```

Comment:
to reduce the complexity of the _load_configuration_impl, can you move this into _process_config_searchpath?

Reply:
done!